charts/spire/charts/spire-agent/README.md

```diff
@@ -109,3 +109,6 @@ A Helm chart to install the SPIRE agent.
 | `experimental.enabled`                            | Allow configuration of experimental features                                                                                                                                                      | `false`                                                                          |
 | `experimental.syncInterval`                       | Sync interval with SPIRE server with exponential backoff                                                                                                                                          | `5s`                                                                             |
 | `experimental.featureFlags`                       | List of developer feature flags                                                                                                                                                                   | `[]`                                                                             |
+| `sockets.hostBasePath`                            | Path on which the agent socket is made available when admin.expose is true                                                                                                                        | `/run/spire/agent/sockets`                                                       |
+| `sockets.admin.enabled`                           | Enable the admin socket. Useful for admin tasks or the Delegated Identity API.                                                                                                                    | `false`                                                                          |
+| `sockets.admin.mountOnHost`                       | Enable the admin socket to be visible on the host.                                                                                                                                                | `false`                                                                          |
```

Comment:
This seems to control the placement of the socket on the host using hostPath semantics.  I'm not sure if we want to support hostPath mounting, as the only client that can support communicating to the socket is the one embedded in the `spire-agent` executable.

Please considering removing this option / configuration approach.

Reply:
The admin socket is also used for the "delegated identity api". At least one major software is needing to use it. Cilium. hostPath is needed to support their needs.

Reply:
Do we support Cilium at this time?  Why are we complicating this PR with items outside the scope of adding the Agent Admin Socket?

Reply:
> The admin socket is also used for the "delegated identity api". At least one major software is needing to use it. Cilium. hostPath is needed to support their needs.

fwiw, my org also makes use of a host-mounted being available for delegated identity. Maybe that's something expected in flex?

> Why are we complicating this PR with items outside the scope of adding the Agent Admin Socket?

I'm generally all for keeping a set of changes as simple as possible, as it has benefits to commit history and reviewer burden, and meshes well with keeping issues as bite-sized as possible for easy wins.
It would be helpful to clarify desired results of #224, and how they seemed to expand due to investigations posted in that issue.
It can also be helpful in online communication where we lack the ability to see tone and expressions to post reasoning/opinions declaratively as much as possible ("I don't think we need foo here, because it is outdated and replaced with bar"), which can then be argued for or against, rather than interrogatively ("Why are we using foo?"), which is more difficult to continue discussion with.

Reply:
@edwbuck Thanks for highlighting this. I talked to @kfox1111 offline and also questioned why this was needed. Upon hearing the cillium use case, it being disabled by default, and the work already being done I determined it was best to leave it in place. Definitely hear the concerns about scope creep, a good reminder to more clearly define requests in the future.

Reply:
@amoore877 Totally understand about small pr's. There are certainly advantages. But we've also had difficulty with pr merge conflicts. So its a difficult balancing act between putting too little in a pr, causing too many pr's, and constantly dealing with merge conflicts vs having too big pr's to easily review. This size pr has been generally a reasonable compromise in the past.

Reply:
@kfox1111 we would have less issue with merge conflicts if the merges were targeted to only the items being changed, as they would be smaller in footprint and less likely to overlap.

What you're offering is too big of a PR with too many PRs and you're getting items from one PR mixed into others.  This is independent of how many PRs are available.  These kinds of "mixing issues" make it difficult to review the PR which is why your PRs take longer to review and get more pushback.

If each PR was small, sweet, and to the point, then reviews would go faster, fewer PRs would remain open, and you'd have less reason to see many PRs with merge conflicts.